optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -839,6 +840,9 @@ public MemberAccessor getConstraintConfigurationMemberAccessor() {
         return constraintConfigurationMemberAccessor;
     }
 
+    /**
+     * @return sometimes null
```

Comment:
Perhaps there would be a better description than `sometimes`?

Reply:
There are basically 2 levels of javadocs:
- Full Javadocs (needed on at least all api package public methods and classes - in theory)
- NonNull Javadocs. Like in Kotlin, it should be clear which methods might return null and which will never return null. The words "never null" and "sometimes null" are used, as a stand in for @NonNull annotations and the like, once we become kotlin-friendly (and java IDE-friendly). Once we use those annotations, those javadocs won't mention it any more.

This impl method falls into the second category - but it really clearly needs to point out that it can return null sometimes.